api/src/workflow/workflow.service.ts

```diff
@@ -1,34 +1,63 @@
-import { Injectable, OnModuleInit } from '@nestjs/common';
-import { Workflow } from "./workflow.entity";
-import { InjectRepository } from "@nestjs/typeorm";
-import { Repository } from "typeorm";
-import { CreateWorkflowDto } from "@stepflow/shared";
+import {
+  Injectable,
+  OnModuleInit,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { Workflow } from './workflow.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateWorkflowDto, CreateWorkflowStepDto } from '@stepflow/shared';
+import { WorkflowStep } from './../wf-step/wf-step.entity';
 
 @Injectable()
 export class WorkflowService implements OnModuleInit {
   constructor(
-    @InjectRepository(Workflow) private readonly workflowRepository: Repository<Workflow>,
+    @InjectRepository(Workflow)
+    private readonly workflowRepository: Repository<Workflow>,
   ) {}
 
   onModuleInit() {
     //generate default data
     this.workflowRepository.save({
-      name: "Basic Workflow 1",
-      description: "",
+      name: 'Basic Workflow 1',
+      description: '',
       steps: [
         {
-          name: "Basic Step 1",
-          description: ""
-        }
-      ]
-    })
+          name: 'Basic Step 1',
+          description: '',
+        },
+      ],
+    });
   }
 
-  create(wf: CreateWorkflowDto): Promise<Workflow> {
-    return this.workflowRepository.save(wf);
+  create(workflowDto: CreateWorkflowDto): Promise<Workflow> {
+    const { name, description, steps } = workflowDto;
+
+    let wokflowSteps = steps.map(step => {
```

Comment:
No need to do this manual mapping.
Use repository method `create` instaed.

```suggestion
    const workflow = this.workflowRepository.create(workflowDto);
```